shaka/src/public/ShakaPlayer.mm

```diff
@@ -136,35 +145,33 @@ @implementation ShakaPlayer
 
 // MARK: setup
 
-- (instancetype)initWithClient:(id<ShakaPlayerClient>)client {
+- (instancetype)initWithError:(NSError *__autoreleasing *)error {
   if ((self = [super init])) {
-    if (![self setClient:client])
+    // Create JS objects.
+    _engine = ShakaGetGlobalEngine();
+    _audio_renderer.reset(new shaka::media::SdlAudioRenderer(""));
+    _media_player.reset(
+            new shaka::media::DefaultMediaPlayer(&_video_renderer, _audio_renderer.get()));
+    _media_player->AddClient(&_client);
+
+    // Set up player.
+    _player.reset(new shaka::Player(_engine.get()));
+    const auto initResults = _player->Initialize(&_client, _media_player.get());
+    if (initResults.has_error()) {
+      *error = [[ShakaPlayerError alloc] initWithError:initResults.error()];
```

Comment:
`error` can be null.  So you need to check before assigning.  If it is null, you should log the error:

```cpp
if (error)
  *error = ...;
else
  LOG(ERROR) << "Error creating player: " << initResults.error().message;
```

Reply:
You are right, there you have it. Thanks!